### PR TITLE
[Quest] fix logic and messages for quest Your Crystal Ball

### PR DESCRIPTION
**Pull request by jamesbradleym**
**_I affirm:_**
- [x] I understand that if I do not agree to the following points by completing the checkboxes my PR will be ignored.
- [x] I understand I should leave resolving conversations to the LandSandBoat team so that reviewers won't miss what was said.
- [x] I have read and understood the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/LandSandBoat/server/blob/base/CODE_OF_CONDUCT.md).
- [x] I have _**tested my code and the things my code has changed**_ since the last commit in the PR and will test after any later commits.

## What does this pull request do?

Closes: https://github.com/HorizonFFXI/HorizonXI-Issues/issues/1841
[Capture](https://www.dropbox.com/scl/fo/guwqlw1l5xlpcnaboho25/h?rlkey=9yfgv7pozhmc8nbo6vtg252c0&dl=0)

Updates the logic gates for Rockwell to prevent odd behavior:
No longer gives player a divination sphere if they interact with Rockwell before trading.
Only gives divination sphere when a lens has been placed in the rockwell previously/since last reward.
Adds trade check that matches trade dialogue ("You cannot submerge more than one Ahriman lens at a time.")

Updates message + item references so interaction dialogue includes item text.
Adds progress() to message so default messages are not called.

Updates default message to pool of water
Adds follow up text about fortune after quest completion.

## Steps to test these changes

!addquest 3 9
!additem 557 (ahriman lens)

interact with Rockwell

## NOTE

Difference between retail and era: Submersion wait in era is JST Midnight, current retail is ~ 30seconds

### CODE REVIEW

File: scripts/zones/Maze_of_Shakhrami/npcs/Rockwell.lua [hunk 1]
```diff
@@ -9,23 +9,31 @@ local ID = zones[xi.zone.MAZE_OF_SHAKHRAMI]
...
 entity.onTrigger = function(player, npc)
     if
         player:getQuestStatus(xi.quest.log_id.JEUNO, xi.quest.id.jeuno.YOUR_CRYSTAL_BALL) == QUEST_ACCEPTED and
-        player:getCharVar('QuestYourCrystalBall_prog') == 1
+        player:getCharVar('QuestYourCrystalBall_prog') ~= 0
     then
-        player:startEvent(52)
-    else
-        player:messageSpecial(ID.text.NOTHING_OUT_OF_ORDINARY)
+        if player:getCharVar('QuestYourCrystalBall_prog') < os.time() then
```
Comment (claywar) [inline scripts/zones/Maze_of_Shakhrami/npcs/Rockwell.lua:30]: This is a bad condition block.  If you're checking less than, there's no need to explicitly check greater than which will also cause a corner case issue of not allowing exactly equal

Comment (jamesbradleym) [reply scripts/zones/Maze_of_Shakhrami/npcs/Rockwell.lua:30]: 100%, not sure where this snuck in but you're right it isn't necessary given the ~=0 condition + < condition.. fixed!

Comment (jamesbradleym) [reply scripts/zones/Maze_of_Shakhrami/npcs/Rockwell.lua:30]: Thanks for the correction :)

**Comment by jamesbradleym** [comment]
@claywar Sorry to bug you on all these, just trying to clear my older PRs 😅 Let me know if you need anything else here.

**Comment by jamesbradleym** [comment]
@claywar Sorry to bump this again, anything else need addressing or is this good to go?

File: scripts/zones/Maze_of_Shakhrami/npcs/Rockwell.lua [hunk 2]
```diff
@@ -9,23 +9,29 @@ local ID = zones[xi.zone.MAZE_OF_SHAKHRAMI]
...
+            end
+        end
     end
 end
 
 entity.onTrigger = function(player, npc)
     if
-        player:getQuestStatus(xi.questLog.JEUNO, xi.quest.id.jeuno.YOUR_CRYSTAL_BALL) == xi.questStatus.QUEST_ACCEPTED and
-        player:getCharVar('QuestYourCrystalBall_prog') == 1
+        player:getQuestStatus(xi.quest.log_id.JEUNO, xi.quest.id.jeuno.YOUR_CRYSTAL_BALL) == xi.questStatus.QUEST_ACCEPTED and
```
Comment (claywar) [inline scripts/zones/Maze_of_Shakhrami/npcs/Rockwell.lua]: Please retest your code. This is not valid.

Comment (jamesbradleym) [reply scripts/zones/Maze_of_Shakhrami/npcs/Rockwell.lua]: Bah! Missed updating to xi.questLog.JEUNO here. Thanks for catching!!

File: scripts/zones/Lower_Jeuno/npcs/Kurou-Morou.lua [hunk 3]
```diff
@@ -96,7 +98,8 @@ entity.onEventFinish = function(player, csid, option, npc)
         player:addTitle(xi.title.FORTUNE_TELLER_IN_TRAINING)
         player:addFame(xi.fameArea.JEUNO, 30)
         player:tradeComplete()
-        player:completeQuest(xi.questLog.JEUNO, xi.quest.id.jeuno.YOUR_CRYSTAL_BALL)
+        player:completeQuest(xi.quest.log_id.JEUNO, xi.quest.id.jeuno.YOUR_CRYSTAL_BALL)
```
Comment (claywar) [inline scripts/zones/Lower_Jeuno/npcs/Kurou-Morou.lua]: Same here

File: scripts/zones/Maze_of_Shakhrami/npcs/Rockwell.lua [hunk 4]
```diff
@@ -9,23 +9,29 @@ local ID = zones[xi.zone.MAZE_OF_SHAKHRAMI]
 local entity = {}
 
 entity.onTrade = function(player, npc, trade)
-    if
-        player:getQuestStatus(xi.questLog.JEUNO, xi.quest.id.jeuno.YOUR_CRYSTAL_BALL) == xi.questStatus.QUEST_ACCEPTED and
-        npcUtil.tradeHas(trade, xi.item.AHRIMAN_LENS)
-    then
-        player:setCharVar('QuestYourCrystalBall_prog', 1)
-        player:confirmTrade(trade)
+    if player:getQuestStatus(xi.quest.log_id.JEUNO, xi.quest.id.jeuno.YOUR_CRYSTAL_BALL) == xi.questStatus.QUEST_ACCEPTED then
```
Comment (claywar) [inline scripts/zones/Maze_of_Shakhrami/npcs/Rockwell.lua]: And here

**Comment by jamesbradleym** [comment]
Closing this PR to reduce open PRs and prevent staleness—will revisit.

**Pull request closed without merging**